OpenRA.Game/Map/Map.cs

```diff
@@ -838,6 +840,14 @@ bool ContainsAllProjectedCellsCovering(MPos uv)
 			if (Grid.MaximumTerrainHeight == 0)
 				return Bounds.Contains(uv.U, uv.V);
 
+			// PERF: Most cells lie within a region where no matter their height,
+			// all possible projected cells would remain in the map area.
+			// For these, we can do a fast-path check.
+			if (projectionSafeBounds.Contains(uv.U, uv.V))
```

Comment:
It looks like this supersedes the `Grid.MaximumTerrainHeight == 0` check above without any additional performance overheads, so we can now remove that?